geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/AccumuloDataStore.scala

```diff
@@ -215,41 +216,35 @@ class AccumuloDataStore(val connector: Connector,
   /**
    * Read Record table name from store metadata
    */
-  def getRecordTableForType(featureType: SimpleFeatureType): String =
-    getRecordTableForType(featureType.getTypeName)
+  def getRecordTable(featureType: SimpleFeatureType): String = getRecordTable(featureType.getTypeName)
 
   /**
    * Read Record table name from store metadata
    */
-  def getRecordTableForType(featureName: String): String =
-    metadata.readRequired(featureName, RECORD_TABLE_KEY)
+  def getRecordTable(featureName: String): String = metadata.readRequired(featureName, RECORD_TABLE_KEY)
 
   /**
    * Read SpatioTemporal Index table name from store metadata
    */
-  def getSpatioTemporalIdxTableName(featureType: SimpleFeatureType): String =
-    getSpatioTemporalIdxTableName(featureType.getTypeName)
+  override def getSpatioTemporalTable(featureType: SimpleFeatureType): String =
+    getSpatioTemporalTable(featureType.getTypeName)
 
   /**
    * Read SpatioTemporal Index table name from store metadata
    */
-  def getSpatioTemporalIdxTableName(featureName: String): String =
-    if (getGeomesaVersion(featureName) > 0) {
-      metadata.readRequired(featureName, ST_IDX_TABLE_KEY)
-    } else {
-      catalogTable
-    }
+  def getSpatioTemporalTable(featureName: String): String =
+    metadata.readRequired(featureName, ST_IDX_TABLE_KEY)
```

Comment:
Yeah, but it's already broken due to schema_changes, with no upgrade path available... Is that something we need to worry about?